homework9/hw2.py

```diff
@@ -0,0 +1,37 @@
+"""
+Write a context manager, that suppresses passed exception.
+Do it both ways: as a class and as a generator.
+
+# >>> with supressor(IndexError):
+# ...    [][2]
+
+"""
+from contextlib import contextmanager
+
+
+@contextmanager
+def supressor(exception):
+    try:
+        yield
+    except exception:
+        ...  # Catch behaviour
+    finally:
```

Comment:
Do you really need a finally block here?

Reply:
Remove it